src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -1318,8 +1361,20 @@ trait ChoosesMode {
   }
 
   when(FinishingModeChoice, stateTimeout = Duration.Zero) { case Event(StateTimeout, data: ChoosesModeData) =>
-    val chosenTrip = data.pendingChosenTrip.get
+    val pendingTrip = data.pendingChosenTrip.get
     val (tick, triggerId) = releaseTickAndTriggerId()
+    val chosenTrip =
+      if (
+        pendingTrip.tripClassifier.isTransit
+        && pendingTrip.legs.head.beamLeg.startTime > tick
+      ) {
+        //we need to start trip as soon as our activity finishes (current tick) in order to
+        //correctly show waiting time for the transit in the OD skims
+        pendingTrip.updatePersonalLegsStartTime(tick)
```

Comment:
Just out of curiosity--would we expect this to give different behavior if instead of the tick we gave the previous activity end time? Would that be more robust to parallelism issues and/or introduce other problems?

Reply:
This tick is the same as the Activity end time. But we could check that the Activity end time is not in the past and just use it here. I'll change it. I can imagine that the Mode Choice may happen not at the activity end time but at some time before that. In this case, using activity end time is better. But we shouldn't use a tick in the past here because it causes issues with the scheduler.